File: components/contracts/cpp/marketplace/src/propwroff.cpp
/**
 * @brief Backend / админ выносит проект списания скоропорта (Story 8.1, p.mkt.wroff).
 *
 * Без ledger2-операций. Создаётся writeoff_proposal в draft со списком items;
 * total_amount = Σ items.amount.
 *
 * Guards:
 *  - actor backend (auth coopname).
 *  - items.size() > 0.
 *  - Все items.amount > 0 в _root_govern_symbol.
 *
 * @ingroup public_marketplace_actions
 *
 * TODO Story 11.1 (Шаг 6): полная реализация.
 */
void marketplace::propwroff(eosio::name coopname,
                             eosio::name proposed_by,
                             checksum256 proposal_hash,
                             std::vector<wroff_item> items) {
  require_auth(coopname);
  eosio::check(false, "TODO Story 11.1 Шаг 6: propwroff ещё не реализован");
}

File: components/cooptypes/src/contracts/marketplace/actions/propWroff.ts
import * as Permissions from '../../../common/permissions'
import type * as Marketplace from '../../../interfaces/marketplace'
import { Actors } from '../../../common'

/**
 * Backend / админ выносит проект списания скоропорта на повестку совета
 * (Story 8.1, p.mkt.wroff). Создаётся writeoff_proposal в PROPOSED;
 * total_amount = Σ items.amount; все items.executed = false.
 */
export const authorizations = [{ permissions: [Permissions.active], actor: Actors._coopname }] as const

export const actionName = 'propwroff'

/**
 * @interface
 */
export type IPropWroff = Marketplace.IPropWroff
